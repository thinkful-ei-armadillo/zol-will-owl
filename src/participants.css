img {
    width: 40px;
    border-radius: 50%;
    border: 1px solid black;
    background: gray;
}

.participantInfo {
    display: inline-block;
    padding-left: 15px;
}


background: rgb(187, 187, 187);
div.participant {
    margin-top: 20px;
    margin-bottom: 20px; 
}

@media ( min-width : 600px ){
  .participantList {
    width: 30%;
    height: 100vh;
  }
}

