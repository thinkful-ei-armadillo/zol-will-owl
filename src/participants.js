import React from 'react';

function User(props) {

    return (
        <div className='user'>
            
        </div>
    )
}


function Participants(props) {

    return (
    // id
    // name
    // avatar
    // session: true or false
    )

}


export default Participants;