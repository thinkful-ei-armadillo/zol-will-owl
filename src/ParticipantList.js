import React from 'react';
import Participant from './Participant';
import './participants.css';


function ParticipantList(props) {
    return (
        <div className="participantList">
            {props.participants.map(participant => 
                <Participant 
                    key={participant.id}
                    name={participant.name}
                    avatar={participant.avatar}
                    inSession={participant.inSession}
                    onStage={participant.onStage}
                />
            )}
        </div>
        )
}


export default ParticipantList;