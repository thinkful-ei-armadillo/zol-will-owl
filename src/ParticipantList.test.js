import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ParticipantList from './ParticipantList';
import participants from './STORE';


describe('<ParticipantList', () => {
    // smoke test
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ParticipantList participants={participants}/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    // snapshot test
    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<ParticipantList participants={participants}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});