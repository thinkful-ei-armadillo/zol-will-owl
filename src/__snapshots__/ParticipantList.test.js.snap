// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ParticipantList renders the UI as expected 1`] = `
<div
  className="participantList"
>
  <div
    className="participant"
  >
    <img
      alt="Koren Templeton"
      className="avatar"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
    <div>
      <span
        className="name"
      >
        Koren Templeton
      </span>
      <div
        className="green"
      />
      <span
        className="status"
      >
        on stage
      </span>
    </div>
  </div>
  <div
    className="participant"
  >
    <img
      alt="Caty Flucker"
      className="avatar"
      src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
    />
    <div>
      <span
        className="name"
      >
        Caty Flucker
      </span>
      <div
        className="green"
      />
      <span
        className="status"
      >
        in session
      </span>
    </div>
  </div>
  <div
    className="participant"
  >
    <img
      alt="Axe Kubicka"
      className="avatar"
      src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
    />
    <div>
      <span
        className="name"
      >
        Axe Kubicka
      </span>
      <div
        className="gray"
      />
      <span
        className="status"
      >
        in session
      </span>
    </div>
  </div>
  <div
    className="participant"
  >
    <img
      alt="Frank Runciman"
      className="avatar"
      src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
    />
    <div>
      <span
        className="name"
      >
        Frank Runciman
      </span>
      <div
        className="green"
      />
      <span
        className="status"
      >
        in session
      </span>
    </div>
  </div>
  <div
    className="participant"
  >
    <img
      alt="Ashla Attwool"
      className="avatar"
      src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
    />
    <div>
      <span
        className="name"
      >
        Ashla Attwool
      </span>
      <div
        className="green"
      />
      <span
        className="status"
      >
        on stage
      </span>
    </div>
  </div>
</div>
`;
